lib/memstore.js

```diff
@@ -59,18 +59,19 @@ MemoryCookieStore.prototype.findCookies = function(domain, path, cb) {
     };
 
   } else {
-    var paths = permutePath(path) || [path];
     pathMatcher = function matchRFC(domainIndex) {
-      paths.forEach(function(curPath) {
-        var pathIndex = domainIndex[curPath];
-        if (!pathIndex) {
-          return;
-        }
-        for (var key in pathIndex) {
-          results.push(pathIndex[key]);
-        }
-      });
-    };
+       //NOTE: we should use path-match algorithm from S5.1.4 here
+       //(see : https://github.com/ChromiumWebApps/chromium/blob/b3d3b4da8bb94c1b2e061600df106d590fda3620/net/cookies/canonical_cookie.cc#L299)
```

Comment:
This is according to the S5.4.1:
> The user agent MUST use an algorithm equivalent to the following
   algorithm to compute the "cookie-string" from a cookie store and a
   request-uri:
....
*  The request-uri's path path-matches the cookie's path.

BTW, seems like we can remove other matcher. I'll try it and will add another commit in case if  it works.